### ISSUE TITLE
Performance regression in sum_kbn with Julia 1.7.1

### ISSUE THREAD

**Issue by goerch** [JuliaMath/KahanSummation.jl#21]
https://discourse.julialang.org/t/speed-issue-with-kahansummation/75491 indicates a performance regression for sum_kbn on Julia 1.7.1.

**Comment by goerch** [comment]
More specifically the following benchmark

```
using KahanSummation, BenchmarkTools

xs = rand(1_000_000)
@btime sum(xs)
@btime sum_kbn(xs)
```

yields

```
  142.400 μs (1 allocation: 16 bytes)
  1.431 ms (1 allocation: 16 bytes)
```

on Julia 1.8.0 and

```
  149.700 μs (1 allocation: 16 bytes)
  106.113 ms (3999488 allocations: 91.54 MiB)
```

on Julia 1.7.1.

**Comment by goerch** [comment]
JET indicates for Julia 1.7.1

```
julia> JET.@report_opt sum_kbn(xs)
═════ 2 possible errors found ═════
┌ @ c:\Users\Win10\Documents\GitHub\KahanSummation.jl\src\KahanSummation.jl:76 Base.indexed_iterate(it, 1)
│┌ @ tuple.jl:92 Base.iterate(I)
││ runtime dispatch detected: Base.iterate(I::Nothing)
│└───────────────
┌ @ c:\Users\Win10\Documents\GitHub\KahanSummation.jl\src\KahanSummation.jl:76 Base.indexed_iterate(it, 2, _11)
│┌ @ tuple.jl:97 Base.iterate(I, state)
││ runtime dispatch detected: Base.iterate(I::Nothing, state::Int64)
│└───────────────
```

**Comment by goerch** [comment]
After adding a type annotation I see

```
  142.200 μs (1 allocation: 16 bytes)
  1.434 ms (1 allocation: 16 bytes)
```

on Julia 1.7.1.

**Comment by ararslan** [comment]
Fixed by #22 :heart:
